example-paginate-nfts/pages/index.js

```diff
@@ -0,0 +1,124 @@
+import Head from 'next/head';
+import styles from '../styles/Home.module.css';
+import { Metaplex } from '@metaplex-foundation/js';
+import { clusterApiUrl, Connection, PublicKey } from '@solana/web3.js';
+import { useEffect, useState } from 'react';
+
+const connection = new Connection(clusterApiUrl('devnet'));
+const mx = Metaplex.make(connection);
+
+export default function Home() {
+  const [address, setAddress] = useState(
+    'EAqjUWVX2m9fdfGNBzTY5zSiid1Sb9V3x6EL8ssZBTkw',
+  );
+
+  const [nftList, setNftList] = useState(null);
+  const [loading, setLoading] = useState(false);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [currentView, setCurrentView] = useState(null);
+  const perPage = 1;
+
+  useEffect(() => {
+    if (nftList) {
+      const execute = async () => {
+        const startIndex = (currentPage - 1) * perPage;
+        const endIndex = currentPage * perPage;
+        await loadData(startIndex, endIndex);
+        setCurrentView(nftList.slice(startIndex, endIndex));
+        setLoading(false);
+      };
+      execute();
+    }
+  }, [nftList, currentPage]);
+
+  const loadData = async (startIndex, endIndex) => {
+    await nftList.forEach(async (nft, index) => {
+      if (
+        index >= startIndex &&
+        index < endIndex &&
+        nft.metadataTask.isPending()
+      ) {
+        await nft.metadataTask.run();
+      }
+    });
```

Comment:
It would be better to load the metadata of all NFTs on the current page in parallel rather than doing it sequentially.

Something like that should work:

```suggestion
    const nftsToLoad = nftList.filter((nft, index) => {
      return (
          index >= startIndex &&
          index < endIndex &&
          nft.metadataTask.isPending()
        );
    });
    
    const promises = nftsToLoad.map(nft => nft.metadataTask.run());
    await Promise.all(promises);
```